include/deal.II/base/std_cxx20/iota_view.h

```diff
@@ -17,15 +17,22 @@
 
 #include <deal.II/base/config.h>
 
-#include <boost/range/irange.hpp>
-
+#if defined __cpp_lib_ranges && __cpp_lib_ranges >= 201911
+#  include <ranges>
+#else
+#  include <boost/range/irange.hpp>
+#endif
 
 DEAL_II_NAMESPACE_OPEN
 
 namespace std_cxx20
 {
   namespace ranges
   {
+#if defined __cpp_lib_ranges && __cpp_lib_ranges >= 201911
+    template <typename IncrementableType, typename BoundType>
+    using iota_view = std::ranges::iota_view<IncrementableType, BoundType>;
```

Comment:
Wouldn't it be possible to just do
```
  using std::ranges::iota_view;
```
(i.e., a traditional `using` declaration rather than a template-alias)?

Reply:
No, that doesn't work. https://godbolt.org/z/rKH2ur

Reply:
Right, but that's not what I meant. You're using the form
```
  using A = B;
```
which acts like a typedef and requires that `B` is a fully qualified name (unless you put `template` in front of it). But I want the name injection:
```
namespace X{
template <typename T>
struct Foo{};
}
using X::Foo;

int main()
{
  Foo<int> foo;
}
```
which actually compiles: https://godbolt.org/z/4A5yKf

Reply:
Yes, that works.